Makefile

```
@@ -0,0 +1,40 @@
+
+MAKEFILE_ROOT=$(shell pwd)
+
+# install: Ensure that the Argo CD namespace exists, and that CRDs we are using are applied to the current cluster
+install:
+	kubectl create ns argocd || true
+	kubectl apply -f $(MAKEFILE_ROOT)/backend/config/crd/bases
+	kubectl apply -f $(MAKEFILE_ROOT)/backend-shared/config/crd/bases/managed-gitops.redhat.com_operations.yaml
+	kubectl apply -f https://raw.githubusercontent.com/argoproj/argo-cd/release-2.2/manifests/crds/application-crd.yaml
+
+# start: start all the components
+start:
+	~/go/bin/goreman start
+
+# build: build all the components (note: you do not need to do this before running start; 'start' will compile and run)
+build:
+	cd $(MAKEFILE_ROOT)/backend && make build
+	cd $(MAKEFILE_ROOT)/cluster-agent && make build
+
```

```suggestion
build: build-backend build-cluster-agent ## Build all the components - note: you do not need to do this before running start

build-backend: ## Build backend only
	cd $(MAKEFILE_ROOT)/backend && make build

build-cluster-agent: ## Build cluster-agent only
	cd $(MAKEFILE_ROOT)/cluster-agent && make build
```

You can build independently each repo for more flexibility ;)